### PR TITLE
Pin the test set and pytorch

### PR DESCRIPTION
**Pull request by PaliC**
As we near release of our alpha, we should make sure users are all TESTING ON THE SAME TEST SET. This PR adds version control to the torchbench test suite to accomplish this.  

The mechanism to do this is we store the info huggingface needs in constants.py which is
```
HUGGINGFACE_REPO - {HF repo the test set lives in}
TEST_SET_REVISION = {which revision of the test set / hf dataset to use (aka the commit hash)}
TEST_SET_FILE = {filename of the test set}
```

It's the responsibility of the backendbench devs to update the test set and these consts (mostly TEST_SET_REVISION) as we revise / add things to the the test set. 

This PR also pins the version of pytorch backendbench uses for similar reasons

### CODE REVIEW

File: BackendBench/data_loaders.py [hunk 1]
```diff
@@ -151,83 +152,86 @@ def _parse_trace_stream(
...
+        return "parquet"
+    elif source_str.endswith(".txt"):
+        return "trace"
+    else:
+        raise ValueError(
+            f"Cannot auto-detect format for source: {source}. Please specify format explicitly."
+        )
+
+
 def load_ops_from_source(
```
[the remark this answers is not in the document]
Comment (PaliC) [reply BackendBench/data_loaders.py:174]: this is false

File: BackendBench/scripts/main.py [hunk 2]
```diff
@@ -98,7 +97,7 @@ def setup_logging(log_level):
 )
 @click.option(
     "--torchbench-data-path",
-    default=DEFAULT_HUGGINGFACE_URL,
+    default=None,
```
Comment (msaroufim) [inline BackendBench/scripts/main.py:100]: is the idea never to have a default now? In which case maybe should update README.md in some way to tell people what to put in

File: BackendBench/data_loaders.py [hunk 3]
```diff
@@ -151,83 +152,86 @@ def _parse_trace_stream(
...
+    if source is None:
+        # read parquet file from huggingface
+        table = load_dataset(
+            HUGGINGFACE_REPO,
+            data_files=TEST_SET_FILE,
+            revision=TEST_SET_REVISION,
+        )["train"]
     else:
+        # read parquet file directly
         table = pq.read_table(source)
```
Comment (msaroufim) [inline BackendBench/data_loaders.py:232]: how will this line work if source is None?

File: BackendBench/scripts/main.py (hunk 2 above)
Comment (PaliC) [reply BackendBench/scripts/main.py:100]: Yeah I'll clarify it in the readme. Effectively, if left none it'll pull it from huggingface.

Comment (msaroufim) [reply BackendBench/scripts/main.py:100]: So there is a default value haha, just add it, easier than updating README

Comment (PaliC) [reply BackendBench/scripts/main.py:100]: I updated the flag to indicate folks shouldn't touch it. Having access to local datasets ends up still being useful as we are updating things.

**Review by msaroufim — CHANGES_REQUESTED** [review]
This PR doesn't actually do what it sets to accomplish
1. The test_set_file and huggingface_repo constants are not constants they're dynamic and they will use whatever is the latest version 
2. The test set revision is helpful but I'm not sure why it should be checked in the repo with the burden of maintaining it when it can be easily generated when main.py starts

It feels like a significantly simpler solution is to
1. When main.py starts print the hash of the latest commit in the repo 
2. Print a date and time (which solves the reproducibility concerns for test file and hf repo)

**Review by msaroufim — APPROVED** [review]
Actually nvm the names threw me off - I'd be fine merging this with a few tiny improvements
1. Move the constants to data loader for now so it's clear the test set revision is w.r.g to the "dataset" and not the repo
2. Give the conts a better name like LAST_COMMIT_ TO_TORCHBENCH_DATA
3. Please revert changes to the README.md
